ocs_ci/helpers/dr_helpers.py

```diff
@@ -488,39 +489,64 @@ def check_vr_state(state, namespace):
         return False
 
 
-def check_vrg_existence(namespace):
+def check_vrg_existence(namespace, vrg_name=""):
     """
     Check if VRG resource exists in the given namespace
 
     Args:
         namespace (str): the namespace of the VRG resource
+        vrg_name (str): Name of VRG
 
     """
-    vrg_list = (
-        ocp.OCP(kind=constants.VOLUME_REPLICATION_GROUP, namespace=namespace)
-        .get()
-        .get("items")
-    )
+    try:
+
+        vrg_list = (
+            ocp.OCP(
+                kind=constants.VOLUME_REPLICATION_GROUP,
+                namespace=namespace,
+                resource_name=vrg_name,
+            )
+            .get()
+            .get("items")
+        )
+    except CommandFailed as e:
+        if (
+            f'Error from server (NotFound): volumereplicationgroups.ramendr.openshift.io "{vrg_name}" not found'
+            in str(e)
+        ):
+            logger.info(f"VRG {vrg_name} not found in namespace {namespace}.")
+            vrg_list = []
+
     if len(vrg_list) > 0:
         return True
     else:
         return False
 
 
-def check_vrg_state(state, namespace):
+def check_vrg_state(state, namespace, resource_name=None):
     """
     Check if VRG in the given namespace is in expected state
 
     Args:
         state (str): The VRG state to check for (e.g. 'primary', 'secondary')
         namespace (str): the namespace of the VRG resources
+        resource_name (str): Name of VRG resource
 
     Returns:
         bool: True if VRG is in expected state or was deleted, False otherwise
 
     """
-    vrg_obj = ocp.OCP(kind=constants.VOLUME_REPLICATION_GROUP, namespace=namespace)
-    vrg_list = vrg_obj.get().get("items")
+
+    vrg_obj = ocp.OCP(
+        kind=constants.VOLUME_REPLICATION_GROUP,
+        namespace=namespace,
+    )
+    if resource_name:
+        vrg_list = vrg_obj.get(resource_name=resource_name)
```

Comment:
Both variables vrg_list and vrg_list_index must be declared before the if condition so that it will not throw error in Line no 552 and Line no 560.

Reply:
we have else condition where if conditional fails, it will go to else and we are setting value at line 548

Reply:
Got it. Thanks Pratik